sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/PushDownUtils.scala

```diff
@@ -107,14 +107,12 @@ object PushDownUtils extends PredicateHelper {
   }
 
   /**
-   * Pushes down aggregates to the data source reader
+   * Translate aggregate expressions and group by expressions.
    *
-   * @return pushed aggregation.
+   * @return translated aggregation.
    */
-  def pushAggregates(
-      scanBuilder: SupportsPushDownAggregates,
-      aggregates: Seq[AggregateExpression],
-      groupBy: Seq[Expression]): Option[Aggregation] = {
+  def translateAggregation(
```

Comment:
shall we put it in `DataSourceStrategy`?

Reply:
OK